core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java

```diff
@@ -170,9 +170,15 @@ public UUID getUserAnonymizedId()
 
 	public Set<Buddy> getBuddiesRelatedToRemovedUsers()
 	{
+		loadAllBuddyUsers();
 		return buddies.stream().filter(b -> b.getUser() == null).collect(Collectors.toSet());
 	}
 
+	private void loadAllBuddyUsers()
```

Comment:
Maybe the name `loadAllBuddyUsersAtOnce` is more speaking. The same for the other methods below.